robot-server/robot_server/runs/run_data_manager.py

```diff
@@ -148,6 +150,28 @@ def get(self, run_id: str) -> Run:
 
         return _build_run(run_resource, state_summary, current)
 
+    def get_run_labware_definition(self, run_id: str) -> List[LabwareDefinition]:
+        """Get a run's labware definition.
+
+        This method will pull from the current run
+        depending on if `run_id` refers to the current run.
+
+        Args:
+            run_id: The identifier of the run to return.
+
+        Returns:
+            The run's labware definitions.
+
+        Raises:
+            RunNotCurrentError: The given run identifier is not the current run.
+        """
+        if run_id != self._engine_store.current_run_id:
+            raise RunNotCurrentError(
+                f"Cannot get labware definitions of {run_id} because it is not the current run."
+            )
+
+        return self._engine_store.engine.state_view.labware.get_all_labware_definition()
```

Comment:
I agree with this, but let's explain why we're doing it.

```suggestion
        # The database doesn't store runs' loaded labware definitions in a way that we
        # can query quickly. Avoid it by only supporting this on in-memory runs.
        if run_id != self._engine_store.current_run_id:
            raise RunNotCurrentError(
                f"Cannot get labware definitions of {run_id} because it is not the current run."
            )

        return self._engine_store.engine.state_view.labware.get_all_labware_definition()
```